src/Hpack/Config.hs

```diff
@@ -150,11 +153,14 @@ instance FromJSON (CaptureUnknownFields FlagSection) where
 
 getUnknownFields :: forall a. HasFieldNames a => Value -> Proxy a -> [FieldName]
 getUnknownFields v _ = case v of
-  Object o -> unknown
+  Object o -> ignoreUnderscored unknown
     where
       unknown = keys \\ fields
       keys = map T.unpack (HashMap.keys o)
       fields = fieldNames (Proxy :: Proxy a)
+      ignoreUnderscored
+        | warnUnderscoredUnknownFields (Proxy :: Proxy a) = id
+        | otherwise = filter (not . isPrefixOf "_")
```

Comment:
It won't, it will ignore this: https://github.com/sol/hpack/pull/146/files/d10a66db70aff2c225622fcdda929d15d92f7312#diff-af92d09f3a5eb3bf28df295b3a1fa13fR327 as well and I didn't consider that a top-level field. But I really don't have any strong feelings about this. If you think the simplification is worth it, let's do it that way. :-)